graylog2-server/src/main/java/org/graylog2/streams/events/StreamsChangedEvent.java

```diff
@@ -0,0 +1,63 @@
+/**
+ * This file is part of Graylog.
+ *
+ * Graylog is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Graylog is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.graylog2.streams.events;
+
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableSet;
+
+import java.util.Set;
+
+@JsonAutoDetect
+@JsonDeserialize(builder = AutoValue_StreamsChangedEvent.Builder.class)
```

Comment:
I will change this instance because there is only one parameter, true. But in general I will keep using the builder pattern for this because the regular factory methods become really hard to work with once you have some more parameters.

And as said earlier, the builder pattern makes it easier to build up immutable collections. The factory pattern needs explicit conversion of collections to immutable collections to avoid having mutable collections in the value objects. I think we have some AutoValue instances in our code where this is broken.